packages/react-composites/src/composites/CallComposite/CallComposite.tsx

```diff
@@ -215,12 +215,17 @@ export const CallComposite = (props: CallCompositeProps): JSX.Element => {
   } = props;
   useEffect(() => {
     (async () => {
-      await adapter.askDevicePermission({ video: true, audio: true });
-      adapter.queryCameras();
-      adapter.queryMicrophones();
-      adapter.querySpeakers();
+      if (role === 'Consumer') {
```

Comment:
when we stop explicitly passing in our role. How are we going to know our role is consumer so we only ask for audio permissions?

Reply:
I hope I understand the question correctly. The CallComposite can only know the role is Consumer when it is passed as a prop because that role will be used for the PermissionProvider.